taichi/ir/frontend_ir.cpp

```diff
@@ -1018,7 +1025,9 @@ Expr ASTBuilder::expr_alloca() {
 Expr ASTBuilder::make_matrix_expr(const std::vector<int> &shape,
                                   const DataType &dt,
                                   const std::vector<Expr> &elements) {
-  return Expr(std::make_shared<MatrixExpression>(elements, shape, dt));
+  auto mat = Expr(std::make_shared<MatrixExpression>(elements, shape, dt));
+  mat->ret_type = dt;
```

Comment:
Is this necessary? It seems that it has already been covered by https://github.com/taichi-dev/taichi/blob/5774ea62166456a9ec6ce4f69534527ba437bfe8/taichi/ir/frontend_ir.cpp#L434-L440

Reply:
This is a workaround for type checking a FrontendAssignStmt. Since the type info only propagates in constructor, we need its type to propagates the type to the variable being assigned.
https://github.com/taichi-dev/taichi/blob/4b1a619e53ba03cec7b1d2ca4a67139d524cdf30/taichi/ir/frontend_ir.cpp#L27-L33